client/web/src/user/settings/codeHosts/OrgUserNeedsCodeHost.tsx

```diff
@@ -0,0 +1,51 @@
+import React from 'react'
+import { Link } from 'react-router-dom'
+
+import { defaultExternalServices } from '@sourcegraph/web/src/components/externalServices/externalServices'
+import { Container } from '@sourcegraph/wildcard'
+
+import { useExternalServices } from '../../../auth/useExternalServices'
+import { ListExternalServiceFields } from '../../../graphql-operations'
+
+export interface UserContext {
+    id: string
+    username: string
+}
+
+export interface OrgUserNeedsCodeHost {
+    orgName: string
+    orgExternalServices: ListExternalServiceFields[]
+    user: UserContext
+}
+
+export const OrgUserNeedsCodeHost: React.FunctionComponent<OrgUserNeedsCodeHost> = ({
+    orgExternalServices,
+    user,
+    orgName,
+}) => {
+    const { externalServices: userExternalServices } = useExternalServices(user.id)
+    const orgKinds = orgExternalServices.map(service => service.kind)
+    const userKinds = new Set((userExternalServices || []).map(service => service.kind))
+    const userMissing = orgKinds.filter(kind => !userKinds.has(kind)).map(kind => defaultExternalServices[kind].title)
+    return (
+        <>
+            {userMissing.length > 0 && (
+                <Container className="mb-4">
+                    <h3>Just one more step...</h3>
+                    <p>
+                        {
+                            <>
+                                Connect with{' '}
+                                {userMissing.length === 1 ? userMissing[0] : userMissing[0] + ' and ' + userMissing[1]}
+                            </>
+                        }{' '}
+                        to start searching across the {orgName} organization's private repositories on Sourcegraph.
+                    </p>
+                    <Link className="btn btn-primary" to={`/users/${user.username}/settings/code-hosts`}>
+                        Connect with {userMissing.length === 1 ? userMissing[0] : 'code hosts'}
+                    </Link>
+                </Container>
+            )}
+        </>
+    )
```

Comment:
IMO extracting the logic outside of JSX would improve readability of the code. 
E.g. we only want to render something if the user has missing connections.

We can also simplify the JSX a little by extracting the missing code hosts to a string variable.

Lastly, should we use some existing component for an alert here? E.g. an info or warning message?

```suggestion
 if (userMissing.length > 0) {
    const missingString = userMissing.join(' and ')
    return (
        <Container className="mb-4">
                    <h3>Just one more step...</h3>
                    <p>
                        Connect with {missingString} to start searching across the {orgName} private repositories on Sourcegraph.
                    </p>
                    <Link className="btn btn-primary" to={`/users/${user.username}/settings/code-hosts`}>
                        Connect with {missingString}
                    </Link>
                </Container>
            )
        } else {
            return null
        }
```

Reply:
Excalidraw for this https://app.excalidraw.com/s/4Dr1S6qmmY7/6YePyrn2Dmv has a structure with a header, body and link at the bottom, so I used an approach similar to what Quinn did in https://github.com/sourcegraph/sourcegraph/pull/28839/files